pkg/combustion/rpm.go

```diff
@@ -110,3 +104,48 @@ func writeRPMScript(ctx *image.Context, rpmFileNames []string) (string, error) {
 
 	return modifyRPMScriptName, nil
 }
+
+// determine whether RPM configuration is needed
+func skipRPMconfigre(ctx *image.Context) bool {
+	pkg := ctx.ImageDefinition.OperatingSystem.Packages
+
+	if isComponentConfigured(ctx, userRPMsDir) && len(pkg.AddRepos) > 0 {
+		// is RPM configured in 'rpms' directory from a third party repository
+		return false
+	} else if isComponentConfigured(ctx, userRPMsDir) {
+		// is RPM configured as a standalone RPM inside of the 'rpms' directory
+		return false
+	}
+
+	// is package configured for installation from PackageHub
+	if len(pkg.PKGList) > 0 && pkg.RegCode != "" {
+		return false
+	}
+
+	// is package configured for installation from a third party repository
+	if len(pkg.AddRepos) > 0 && len(pkg.PKGList) > 0 {
+		return false
+	}
+
+	return true
+}
+
+// determine whether package/rpm dependency resolution is needed
+func isResolutionNeeded(ctx *image.Context) bool {
```

Comment:
Validation does not handle use-cases when a user provides an RPM and a registration code. This will be fixed.